src/workerd/jsg/jsg.h

```diff
@@ -2608,6 +2608,8 @@ class Lock {
   JsSymbol symbolShared(kj::StringPtr) KJ_WARN_UNUSED_RESULT;
   JsSymbol symbolInternal(kj::StringPtr) KJ_WARN_UNUSED_RESULT;
   JsObject obj() KJ_WARN_UNUSED_RESULT;
+  JsObject obj(
+      kj::ArrayPtr<kj::StringPtr> keys, kj::ArrayPtr<kj::StringPtr> values) KJ_WARN_UNUSED_RESULT;
```

Comment:
I would make the values an array of jsg::JsValue instead of requiring them to be strings.